src/data/finsupp/witnesses.lean

```diff
@@ -0,0 +1,209 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.finsupp.lex
+
+/-!
+#  Witnesses, linear orders and covariant classes in  in finsupps
+-/
+
+variables {α N : Type*}
+
+lemma ne_or_ne_of_ne {a b : α} (c : α) (h : a ≠ b) :
+  a ≠ c ∨ b ≠ c :=
+not_and_distrib.1 $ mt (and_imp.2 eq.substr) h.symm
+
+namespace finsupp
+section decidables
+variables [decidable_eq α] [decidable_eq N]
+
+section diff
+
+section N_has_zero
+variables [has_zero N] {f g : α →₀ N}
+
+/--  `f.diff g` is the `finset` where `f` and `g` differ. -/
+def diff (f g : α →₀ N) : finset α :=
```

Comment:
I don't really agree, especially since @stuart-presnell is in the process of splitting that file.

Reply:
Ok, I will not increase the length of `data.finsupp.basic` with definitions and results that have not been needed so far.

Reply:
I think this is generic enough that it could be useful outside of this file. Also, refactors shouldn't halt the development of files unless the developments directly conflict with the refactors.

Reply:
What's to stop those other files it might be useful in importing this one?

Reply:
> the developments directly conflict with the refactors.

They do; the refactor observes "finsupp.basic" is too big. Adding more new definitions to it rather than in new files makes it bigger, which directly conflicts with the refator task of making it not too big.